packages/itwin/ec3-widget/src/components/LabelAction.tsx

```diff
@@ -30,21 +31,21 @@ export const LabelAction = ({ template, onClose, label, setTemplate }: LabelActi
   const [name, setName] = useState<string>(label?.name ?? "");
   const [itemName, setItemName] = useState<string>(label?.elementNameColumn ?? "UserLabel");
   const [itemQuantity, setItemQuantity] = useState<string>(label?.elementQuantityColumn ?? "");
-  const [selectedMaterial, setSelectedMaterial] = useState<Material | undefined>();
+  const [selectedMaterial, setSelectedMaterial] = useState<EC3ConfigurationMaterial | undefined>();
   const [isLoading, setIsLoading] = useState<boolean>(true);
-  const [materials, setMaterials] = useState<Material[]>(label?.materials.map((x) => { return { nameColumn: x.nameColumn }; }) ?? [{ nameColumn: undefined }]);
+  const [materials, setMaterials] = useState<(EC3ConfigurationMaterial | undefined)[]>(label?.materials ?? [undefined]);
```

Comment:
What's the point of having `undefined` in the array? Why not an empty array in that case?

Reply:
I did that originally, but the materials selection is written in such an odd way that it would require extensive refactoring. That would have to be in another PR.